packages/fxa-settings/src/components/App/index.tsx

```diff
@@ -75,32 +76,41 @@ export const App = ({ flowQueryParams, navigatorLanguages }: AppProps) => {
             <PageSettings path="/" />
             <PageDisplayName path="/display_name" />
             <PageAvatar path="/avatar" />
-            <PageChangePassword path="/change_password" />
             {hasPassword ? (
-              <PageRecoveryKeyAdd path="/account_recovery" />
+              <>
+                <PageChangePassword path="/change_password" />
+                <Redirect
+                  from="/create_password"
+                  to="/settings/change_password"
+                  noThrow
+                />
+                <PageRecoveryKeyAdd path="/account_recovery" />
+                <PageTwoStepAuthentication path="/two_step_authentication" />
+                <Page2faReplaceRecoveryCodes path="/two_step_authentication/replace_codes" />
+              </>
             ) : (
-              <Redirect from="/account_recovery" to="/settings" noThrow />
+              <>
+                <PageCreatePassword path="/create_password" />
+                <Redirect
+                  from="/change_password"
+                  to="/settings/create_password"
+                  noThrow
+                />
+                <Redirect from="/account_recovery" to="/settings" noThrow />
+                <Redirect
+                  from="/two_step_authentication"
+                  to="/settings"
+                  noThrow
+                />
+                <Redirect
+                  from="/two_step_authentication/replace_codes"
+                  to="/settings"
+                  noThrow
+                />
+              </>
             )}
             <PageSecondaryEmailAdd path="/emails" />
```

Comment:
Probably a product question, but should we require users to set a password for adding emails? I'm not sure where to draw the line but in my ideal world we would try to nudge users to create password when doing account level stuff.

Reply:
Good question, I actually have a 1:1 with Vesta later this week and I'll ask her about this. Setting an avatar and display name seems harmless enough to me IMO but I can see what you mean about secondary emails so yeah I'll ask what's ideal. We'd also want to change the copy we've got around "set a password to use certain security features..." (more of a note to self).